ProcessLib/HT/CreatePorousMediaProperties.cpp

```diff
@@ -0,0 +1,96 @@
+#include "MaterialLib/PorousMedium/Storage/createStorageModel.h"
+
+namespace ProcessLib
+{
+namespace HT
+{
+PorousMediaProperties createPorousMediaProperties(
+    MeshLib::Mesh& mesh, BaseLib::ConfigTree const& porous_medium_configs)
+{
+    DBUG("Create PorousMediaProperties.");
+
+    std::vector<Eigen::MatrixXd> intrinsic_permeability_models;
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Porosity>>
+        porosity_models;
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Storage>>
+        storage_models;
+
+    std::vector<int> mat_ids;
+    for (auto const& porous_medium_config :
+         //! \ogs_file_param_special{prj__processes__process__HT__porous_medium__porous_medium}
+         porous_medium_configs.getConfigSubtreeList("porous_medium"))
+    {
+         //! \ogs_file_param_special{prj__processes__process__HT__porous_medium__porous_medium__id}
+        auto const id = porous_medium_config.getConfigAttribute<int>("id");
+        mat_ids.push_back(id);
+
+        auto const& porosity_conf =
+            //! \ogs_file_param_special{prj__processes__process__HT__porous_medium__porous_medium__porosity}
+            porous_medium_config.getConfigSubtree("porosity");
+        porosity_models.emplace_back(
+            MaterialLib::PorousMedium::createPorosityModel(porosity_conf));
+
+        // Parameter for the intrinsic permeability (only one scalar per
+        // element,
+        // i.e., the isotropic case is handled at the moment)
+        auto const& permeability_conf =
+            //! \ogs_file_param_special{prj__processes__process__HT__porous_medium__porous_medium__permeability}
+            porous_medium_config.getConfigSubtree("permeability");
+        intrinsic_permeability_models.emplace_back(
+            MaterialLib::PorousMedium::createPermeabilityModel(
+                permeability_conf));
+
+        // Parameter for the specific storage.
+        auto const& storage_conf =
+            //! \ogs_file_param_special{prj__processes__process__HT__porous_medium__porous_medium__storage}
+            porous_medium_config.getConfigSubtree("storage");
+        storage_models.emplace_back(
+            MaterialLib::PorousMedium::createStorageModel(storage_conf));
+    }
+
+    BaseLib::reorderVector(intrinsic_permeability_models, mat_ids);
+    BaseLib::reorderVector(porosity_models, mat_ids);
+    BaseLib::reorderVector(storage_models, mat_ids);
+
+    std::vector<int> material_ids;
+    auto const& mesh_material_ids =
+        mesh.getProperties().getPropertyVector<int>("MaterialIDs");
+    if (mesh_material_ids)
+    {
+        material_ids.reserve(mesh_material_ids->size());
+        std::copy(mesh_material_ids->cbegin(), mesh_material_ids->cend(),
+                  material_ids.begin());
+    }
+    else
+    {
+        material_ids.resize(mesh.getNumberOfElements());
+        std::fill_n(material_ids.begin(), mesh.getNumberOfElements(), 0);
```

Comment:
fill after resize is redundant. If you want to be sure, pass a required value to resize as second argument.
http://en.cppreference.com/w/cpp/container/vector/resize